### PR TITLE
only build once

### PR DESCRIPTION
**Pull request by matt-chan**

### CODE REVIEW

**Comment by tovrstra** [comment]
Looks pretty good. I think most problems you encounter can be solved by making a debug package (with all the coverage and other debug flags set), installing it and running all tests and linting on it.

The way to do so is to add a line to you recipe to pass a variable to the conda build.sh script that triggers the debug build. See https://conda.io/docs/user-guide/tasks/build-packages/define-metadata.html#use-environment-variables E.g. you can set a `DEBUG_PACKAGE` variable to 1 or something along those lines. It can then be picked up in the build.sh files to perform the compilation with debug settings.

The advantage of such an approach is that all libraries will be found as if you are building a release package, without too much effort, e.g. you won't need `python-qcgrids/setup.cfg`.

That said, you can wonder how in-place builds should be done right?

File: tools/.travis-template.yml [hunk 1]
```diff
@@ -77,70 +78,43 @@ script:
...
+    conda build -q tools/conda.recipe;
+    (cd ${PYDIR}; conda build -q tools/conda.recipe);
+    conda install --use-local ${CONDA_PKG_NAME_CPP} ${CONDA_PKG_NAME_PY};
+    ${CONDA_PREFIX}/libexec/${PROJECT_NAME}/test_${PROJECT_NAME};
+    (cd; nosetests ${PROJECT_NAME} -v --detailed-errors);
+    conda uninstall ${CONDA_PKG_NAME_PY};
+    conda uninstall ${CONDA_PKG_NAME_CPP};
+  else
+    ${DEPENDENCIES};
+    (mkdir debug; cd debug; cmake cmake -DCMAKE_BUILD_TYPE=debug ..; make);
```
Comment (tovrstra) [inline tools/.travis-template.yml]: typo? cmake cmake

Comment (matt-chan) [reply tools/.travis-template.yml]: It's always been there. I'll take it out though.

**Comment by matt-chan** [comment]
Yes, I'd be happier with the conda debug builds too. We can replicate on our own machines easier that way. Also I think we can take the packages and send them around. The cost is a slightly longer build I think.

@tovrstra can you take a look at the .cardboardlint recipe if you get time? It's giving an error and I don't know why. I think something changed in the most recent version of cardboardlint?

**Comment by tovrstra** [comment]
This should fix it. The latest cardboardlinter fails when unknown config tags are used. At some point I made the doxygen linter simpler to use (less config).

**Comment by tovrstra** [comment]
I'll merge because this is already a very nice improvement! We can always refine later.

**Pull request merged**